org.eclipse.scanning.event.queues/src/org/eclipse/scanning/event/queues/processes/MonitorAtomProcess.java

```diff
@@ -64,131 +69,134 @@
-		ILazyWriteableDataset writer = new LazyWriteableDataset(name, Dataset.FLOAT, shape, shape, shape, null); // DO NOT COPY!
-		file.createData(par, writer);
-		broadcast(Status.RUNNING, 20.0, "Lazydataset created.");
-		
-		// Write Value
-		SliceND slice = SliceND.createSlice(writer, new int[]{0}, new int[]{1});
-		IDataset toWrite = DatasetFactory.createFromObject(scannable.getPosition());
-		writer.setSlice(new IMonitor.Stub(), toWrite, slice); 
-		broadcast(Status.RUNNING, 50.0, "Slice written.");
-			
-		file.close();
+	
+	private void cleanUp(NexusFile nxsFile) {
+		final String path = nxsFile.getFilePath();
+		final File nxsFSObj = new File(path);
 		
-		return "/entry1/instrument/"+name;
-	}
-
-
-	private File createNewTemporaryFile() {
-		IFilePathService service = ServicesHolder.getFilePathService();
-		File dir = new File(service.getTempDir());
-		final String name = UniqueUtils.getSafeName(getQueueBean().getName());
-		return UniqueUtils.getUnique(dir, name, "nxs");
-	}
-
-	@Override
-	protected void postMatchAnalysis() throws EventException, InterruptedException {
-		try {
-			postMatchAnalysisLock.lockInterruptibly();
-
-			if (isTerminated()) {
-				broadcast("Move aborted before completion (requested).");
-				return;
+		if (nxsFSObj.exists()) {
+			try {
+				nxsFile.close();
+			} catch (NexusException nxEx) {
+				logger.debug("Failed to close Nexus file {} during cleanUp", path);
 			}
-
-			executed = true;
-			if (queueBean.getPercentComplete() >= 99.5 && !terminated) {
-				//Clean finish
-				broadcast(Status.COMPLETE, 100d, "Device move(s) completed.");
-			} else {
-				//Scan failed
-				//TODO Set message? Or is it set elsewhere?
-				broadcast(Status.FAILED);
-			} 
-		} finally {
-			//And we're done, so let other processes continue
-			executionEnded();
-
-			postMatchAnalysisLock.unlock();
-
-			/*
-			 * N.B. Broadcasting needs to be done last; otherwise the next 
-			 * queue may start when we're not ready. Broadcasting should not 
-			 * happen if we've been terminated.
-			 */
-			if (!isTerminated()) {
-				broadcast();
+			
+			boolean deletedNexusFile = nxsFSObj.delete();
+			if (!deletedNexusFile) {
+				logger.warn("Failed to delete NeXus file{} during cleanup", path);
 			}
 		}
 	}
-	
-	@Override
-	public void doTerminate() throws EventException {
-		try {
-			//Reentrant lock ensures execution method (and hence post-match 
-			//analysis) completes before terminate does
-			postMatchAnalysisLock.lockInterruptibly();
 
-			//TODO Additional abort action, not handled as part of run()?
-			terminated = true;
+	@Override
+	protected void postMatchAnalysis() throws EventException, InterruptedException {
+		if (isTerminated()) {
+			queueBean.setMessage("Get value of '"+queueBean.getMonitor()+"' aborted (requested)");
+			logger.debug("'"+bean.getName()+"' was requested to abort");
+			return;
+		} else if (queueBean.getPercentComplete() >= 99.5 && !terminated) {
```

Comment:
In another place we have  "99.49) {//99.49 to catch rounding errors" Why not here?
Also those blocks seem to be if (terminaled) ... else if (>99.5) ... else ...  This logic should appear once ideally only the three actions ... being different.